src/app/services/zap.service.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable } from "@angular/core";
+import { NostrEventDocument, Zapper, ParsedZap } from "./interfaces";
+var lightningPayReq = require('light-bolt11-decoder')
+import { sha256 } from "@noble/hashes/sha256";
+import { Utilities } from './utilities';
+import { ProfileService } from './profile';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class ZapService {
+
+
+    constructor(private util: Utilities) { }
+
+
+    parseZap(zapEvent: NostrEventDocument): ParsedZap {
+        const { amount, hash } = this.getInvoice(zapEvent);
+        const zapper = hash ? this.getZapper(zapEvent, hash) : ({ isValid: false, content: "" } as Zapper);
+        const e = this.zappedAuthor(zapEvent);
+        const p = this.zappedPost(zapEvent);
+        debugger
+
+        return {
+            id: zapEvent.id,
+            e: e != undefined ? e : undefined,
+            p: p != undefined ? p : "",
+            amount: Number(amount) / 1000,
```

Comment:
What is the amount in this case and why does it have to be divided by 1000?

Reply:
Because the Lightning Network uses millisatoshi as a unit of account, Converting to millisatoshi allows for compatibility with the Lightning Network's payment channel.  

Your changes were very helpful in improving the code btw, thanks

Reply:
Thanks, added this to make it clear: https://github.com/block-core/blockcore-notes/commit/29593d117acc42e8b04559e31281444a0b67b175